src/v/storage/parser.cc

```diff
@@ -185,7 +190,10 @@ ss::future<result<stop_parser>> continuous_batch_parser::consume_one() {
         if (st.value() == stop_parser::yes) {
             return ss::make_ready_future<result<stop_parser>>(st.value());
         }
-        return consume_records();
+        return consume_records().finally([this] {
+            // reset header
+            add_bytes_and_reset();
```

Comment:
from a design perspective. i wonder if use the 'peek()' for header bytes? from the underlying stream? i guess we can't peek the stream?

Reply:
exactly, there is no `peek`